app/gui2/src/main.ts

```diff
@@ -1,38 +1,76 @@
+import { urlParams } from '@/util/urlParams'
+import {
+  checkMinSupportedVersion as checkMinimumSupportedVersion,
+  displayDeprecatedVersionDialog,
+} from '@/util/version'
+import { run as runDashboard } from 'enso-authentication'
+import { isOnLinux } from 'enso-common/src/detect'
 import 'enso-dashboard/src/tailwind.css'
+import { decodeQueryParams } from 'lib0/url'
 
-const INITIAL_URL_KEY = `Enso-initial-url`
+const vueAppEntry = import('./createApp')
 
-import * as dashboard from 'enso-authentication'
-import { isMac } from 'lib0/environment'
-import { decodeQueryParams } from 'lib0/url'
+const INITIAL_URL_KEY = `Enso-initial-url`
+const SCAM_WARNING_TIMEOUT = 1000
 
 const params = decodeQueryParams(location.href)
-
-// Temporary hardcode
-const config = {
-  supportsLocalBackend: true,
-  supportsDeepLinks: isMac,
-  shouldUseAuthentication: false,
-  projectManagerUrl: PROJECT_MANAGER_URL,
-  initialProjectName: params.project ?? null,
-}
+const initialProjectName = params.project ?? null
 
 let unmount: null | (() => void) = null
 let runRequested = false
 
+function printScamWarning() {
+  const headerCss = `
+    color: white;
+    background: crimson;
+    display: block;
+    border-radius: 8px;
+    font-weight: bold;
+    padding: 10px 20px 10px 20px;
+  `
+    .trim()
+    .replace(/\n\s+/, ' ')
+  const headerCss1 = headerCss + ' font-size: 46px;'
+  const headerCss2 = headerCss + ' font-size: 20px;'
+  const msgCSS = 'font-size: 16px;'
+
+  const msg1 =
+    'This is a browser feature intended for developers. If someone told you to ' +
+    'copy-paste something here, it is a scam and will give them access to your ' +
+    'account and data.'
+  const msg2 = 'See https://enso.org/selfxss for more information.'
+  console.log('%cStop!', headerCss1)
+  console.log('%cYou may be the victim of a scam!', headerCss2)
+  console.log('%c' + msg1, msgCSS)
+  console.log('%c' + msg2, msgCSS)
+}
+
+printScamWarning()
+let scamWarningHandle = 0
+window.addEventListener('resize', () => {
+  window.clearTimeout(scamWarningHandle)
+  scamWarningHandle = window.setTimeout(printScamWarning, SCAM_WARNING_TIMEOUT)
+})
+
 export interface StringConfig {
   [key: string]: StringConfig | string
 }
 
-const vueAppEntry = import('./createApp')
-
 async function runApp(config: StringConfig | null, accessToken: string | null, metadata?: object) {
   runRequested = true
   const { mountProjectApp } = await vueAppEntry
   if (runRequested) {
     unmount?.()
-    const app = mountProjectApp({ config, accessToken, metadata })
-    unmount = () => app.unmount()
+    const options = appOptions.clone()
+    const didOptionsLoad = options.loadAllAndDisplayHelpIfUnsuccessful([urlParams()])
+    if (didOptionsLoad && !(await checkMinimumSupportedVersion(options))) {
+      displayDeprecatedVersionDialog()
+    } else {
+      const app = mountProjectApp({ config, accessToken, metadata })
+      unmount = () => app.unmount()
```

Comment:
IMO the app should always be mounted, but the app root would decide whether or not to display a proper project page, or a warning. That way we can make those warning/help pages in Vue.